src/API/JobMetadataBuilder.cs

```diff
@@ -0,0 +1,36 @@
+/*
+ * Apache License, Version 2.0
+ * Copyright 2021 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Ardalis.GuardClauses;
+using System.Collections.Generic;
+
+namespace Nvidia.Clara.DicomAdapter.API
+{
+    public class JobMetadataBuilder : Dictionary<string, string>
+    {
+        public void AddSourceName(string value)
+        {
+            Guard.Against.NullOrWhiteSpace(value, nameof(value));
+            this["Source"] = value;
+        }
+
+        public void AddInstanceCount(int instanceCount)
+        {
+            this["Instances"] = $"{instanceCount}";
```

Comment:
careful here, this value could be present `1000` as `1,000` or `1.000` depending on culture. is likely better to use `$"{instanceCount:D}` which guarantees `1000`

Reply:
Will fix in upcoming PR